src/main/java/org/broadinstitute/dsm/jobs/TestBostonUPSTrackingJob.java

```diff
@@ -180,61 +202,59 @@ private static void updateTrackingInfo(String statusType,
-                }
+        try (PreparedStatement stmt = conn.prepareStatement(query)) {
+            stmt.setString(1, upsUpdate);
+            stmt.setString(2, date);
+            stmt.setString(3, trackingId);
+            stmt.setString(4, kit.getExternalOrderNumber());
+            int r = stmt.executeUpdate();
+            if (r != 2) {//number of subkits
+                logger.error("Update query for UPS tracking updated " + r + " rows! with tracking/return id: " + trackingId + " for kit w/ external order number "+ kit.getExternalOrderNumber());
+            }
 
-                if (!isReturn) {
-                    if (shouldTriggerEventForKitOnItsWayToParticipant(statusType, oldType)) {
-                        KitDDPNotification kitDDPNotification = KitDDPNotification.getKitDDPNotification(conn,SQL_SELECT_KIT_FOR_NOTIFICATION_EXTERNAL_SHIPPER + SELECT_BY_TRACKING_NUMBER, new String[] { DELIVERED, trackingId }, 2);//todo change this to the number of subkits but for now 2 for test boston works
-                        if (kitDDPNotification != null) {
-                            logger.info("Triggering DDP for kit going to participant with external order number: " + kit.getExternalOrderNumber());
-                            EventUtil.triggerDDP(conn, kitDDPNotification);
-                        }
-                        else {
-                            logger.error("delivered kitDDPNotification was null for " + kit.getExternalOrderNumber());
-                        }
+            if (!isReturn) {
+                if (shouldTriggerEventForKitOnItsWayToParticipant(statusType, oldType)) {
+                    KitDDPNotification kitDDPNotification = KitDDPNotification.getKitDDPNotification(conn,SQL_SELECT_KIT_FOR_NOTIFICATION_EXTERNAL_SHIPPER + SELECT_BY_TRACKING_NUMBER, new String[] { DELIVERED, trackingId }, 2);//todo change this to the number of subkits but for now 2 for test boston works
+                    if (kitDDPNotification != null) {
+                        logger.info("Triggering DDP for kit going to participant with external order number: " + kit.getExternalOrderNumber());
+                        EventUtil.triggerDDP(conn, kitDDPNotification);
+                    }
+                    else {
+                        logger.error("delivered kitDDPNotification was null for " + kit.getExternalOrderNumber());
                     }
+                }
 
+            }
+            else { // this is a return
+                if (earliestInTransitTime != null && !kit.isCEOrdered()) {
+                    // if we have the first date of an inbound event, create an order in CE
+                    // using the earliest date of inbound event
+                    orderRegistrar.orderTest(DSMServer.careEvolveAuth, kit.getHRUID(), kit.getMainKitLabel(), kit.getExternalOrderNumber(), earliestInTransitTime);
```

Comment:
@pegahtah is this and L240 the only places where it is important that kitLabel is not `_1`

Reply:
Yes rest share the same info only where kit label is  needed for a method we need it to be the main one